libs/dh/core/shell/src/lib/dh-core-shell.module.ts

```diff
@@ -45,25 +58,47 @@ const routes: Routes = [
           import('@energinet-datahub/dh/metering-point/shell').then(
             (esModule) => esModule.DhMeteringPointShellModule
           ),
+        canActivate: [MsalGuard],
       },
     ],
   },
-  // { path: '**', component: PageNotFoundComponent },
+  // Used by MSAL (B2C)
+  { path: 'state', redirectTo: '', pathMatch: 'full' },
 ];
 
 @NgModule({
   exports: [RouterModule],
   imports: [
-    DhCoreShellScam,
-    HttpClientModule,
     DhApiModule.forRoot(),
+    DhCoreShellScam,
     DhTranslocoModule.forRoot(),
+    HttpClientModule,
+    MsalModule,
     DhConfigurationLocalizationModule.forRoot(),
     RouterModule.forRoot(routes, {
       anchorScrolling: 'enabled',
-      initialNavigation: 'enabledNonBlocking',
+      // Don't perform initial navigation in iframes or popups
+      initialNavigation:
+        BrowserUtils.isInIframe() && BrowserUtils.isInPopup()
+          ? 'disabled'
+          : 'enabled',
```

Comment:
suggestion: The `enabled` keyword is [deprecated](https://angular.io/api/router/InitialNavigation) since Angular v11. Let's use `enabledNonBlocking` instead.